File: ai/src/main/java/org/pingel/forms/DoubleValue.java
/*
 * Created on Jun 2, 2005
 *
 */
package org.pingel.forms;

import org.pingel.gestalt.core.Form;
import org.pingel.gestalt.core.Name;
import org.pingel.gestalt.core.SimpleForm;
import org.pingel.type.Reals;

public class DoubleValue  {

    public double val;

    public Form createDoubleValue(double val)
    {
        this.val = val;
        return new SimpleForm(new Name(val + ""));
    }
    
    public boolean equals(Reals other) {
    	
        if( other instanceof DoubleValue ) {
            DoubleValue dv = (DoubleValue) other;
            return (dv != null) && this.val == dv.val;
        }
        else {
            return false;
        }
    }

    public int compareTo(Form other)
    {
        if( other instanceof DoubleValue ) {
            DoubleValue dv = (DoubleValue) other;
            return new Double(val).compareTo(dv.val);
        } 
        else {
            return -1;
        }

    }

    public String toLaTeX()
    {
        return val + "";
    }
}

File: ai/src/main/scala/org/pingel/forms/DoubleValue.scala
/*
 * Created on Jun 2, 2005
 *
 */
package org.pingel.forms

import org.pingel.gestalt.core.Form
import org.pingel.gestalt.core.Name
import org.pingel.gestalt.core.SimpleForm
import org.pingel.ptype.PReals

case class DoubleValue(value: Double)  extends SimpleForm(new Name(value + "")) {
  
//    def createDoubleValue(pValue: Double) = {
//        new SimpleForm(new Name(pValue + ""))
//    }
    
    def equals(other: PReals) = other match {
      case dv: DoubleValue => (dv != null) && this.value == dv.value
      case _ => false
    }

    def compareTo(other: Form) = other match {
      case dv: DoubleValue => DoubleValue(value).compareTo(dv.value)
      case _ => -1
    }

    def toLaTeX() = value + ""

}
